src/server/connection.py

```diff
@@ -120,13 +125,29 @@ def __str__(self) -> str:
 
 
 class PeerConnections:
-    def __init__(self, all_connections: List[ChiaConnection] = []):
+    def __init__(
+        self,
+        local_type: NodeType,
+        root_path,
+        config: Dict,
```

Comment:
How many config items do we need? If it's just one or two, can we pass in just those that are used? Passing around opaque config dictionaries is something I'd like to clean up, and eventually use fully-typed configuration structs, like what's been done with `ConsensusConstants`.